intf/genredexpr.mli

```diff
@@ -52,7 +52,7 @@ type ('a,'b,'c) red_expr_gen =
 type ('a,'b,'c) may_eval =
   | ConstrTerm of 'a
   | ConstrEval of ('a,'b,'c) red_expr_gen * 'a
-  | ConstrContext of (Loc.t * Id.t) * 'a
+  | ConstrContext of Id.t Loc.located * 'a
```

Comment:
Good point, the situation wrt id is still too inconsistent, added to the todo list.